data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaSourceCustomConsumer.java

```diff
@@ -438,4 +450,26 @@ public void onPartitionsRevoked(Collection<TopicPartition> partitions) {
     private long getPartitionEpoch(final TopicPartition topicPartition) {
         return ownedPartitionsEpoch.getOrDefault(topicPartition, 0L);
     }
+
+    final void dumpTopicPartitionOffsets(final Collection<TopicPartition> partitions) {
+        try {
+            final Map<TopicPartition, OffsetAndMetadata> committedOffsets = consumer.committed(new HashSet<>(partitions));
+            final Map<TopicPartition, Long> beginningOffsets = consumer.beginningOffsets(partitions);
+            final Map<TopicPartition, Long> endOffsets = consumer.endOffsets(partitions);
+            for (TopicPartition topicPartition : partitions) {
+                final OffsetAndMetadata offsetAndMetadata = committedOffsets.get(topicPartition);
+                LOG.info("Assigned partition {} with offsets: beginningOffset: {}, endOffset: {}, committedOffset: {}",
```

Comment:
Minor: this might be better off as a debug log rather than info

Reply:
this is very crucial log for debugging, I wish we had it from the beginning. If debug shows up by default then I can change it to debug level else leave it as INFO.